src/Paket.Core/Versioning/Requirements.fs

```diff
@@ -791,25 +791,30 @@ type PackageRequirement =
     member this.Depth = this.Graph.Length
 
     static member Compare(x,y,startWithPackage:PackageFilter option,boostX,boostY) =
-        if x = y then 0 else
-        seq {
-            yield compare
-                (not x.VersionRequirement.Range.IsGlobalOverride,x.Depth)
-                (not y.VersionRequirement.Range.IsGlobalOverride,y.Depth)
-            yield match startWithPackage with
+        if obj.ReferenceEquals(x, y) then 0 else
```

Comment:
Not necessary, but performance improvement.
It first compared the whole objects, then compared each member again.

Profiler showed a small improvement.